src/se/terhol/pisemka23/Circuit.java

```diff
@@ -0,0 +1,69 @@
+package se.terhol.pisemka23;
+
+/**
+ * @author Tereza Holm
+ */
+public class Circuit implements ICircuit {
+    private int numSockets;
+    private int maxPower;
+    private ElectricDevice[] connectedDevices;
+
+    public Circuit(int numSockets, int maxPower) {
+        if (numSockets <= 0 || maxPower <= 0) {
+            throw new IllegalArgumentException();
+        }
+        this.numSockets = numSockets;
+        this.maxPower = maxPower;
+        connectedDevices = new ElectricDevice[numSockets];
+    }
+
+    @Override
+    public CircuitBreaker getBreaker() {
+        return new CircuitBreaker(maxPower, connectedDevices);
+    }
+
+    @Override
+    public int isConnected(ElectricDevice device) {
+        int isConnected = -1;
+        for (int i = 0; i < numSockets; i++) {
+            if (device.equals(connectedDevices[i])) {
+                isConnected = i;
+            }
+        }
+        return isConnected;
+    }
+
+    @Override
+    public int getFreeSocket() {
+        int freeSocket = -1;
+        for (int i = 0; i < numSockets; i++) {
```

Comment:
We should probably extract this to a private method so we can avoid `break` in the cycle

Reply:
Done